tests/unit/records/test_records_tasks.py

```diff
@@ -0,0 +1,136 @@
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Test Zenodo records tasks."""
+from copy import deepcopy
+from datetime import datetime
+import uuid
+
+from invenio_cache import current_cache
+from invenio_pidrelations.contrib.versioning import PIDVersioning
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+
+from zenodo.modules.records.api import ZenodoRecord
+from zenodo.modules.records.minters import zenodo_record_minter
+from zenodo.modules.records.tasks import schedule_update_datacite_metadata
+
+
+def test_datacite_update(mocker, db, minimal_record):
+    dc_mock = mocker.patch(
+        'invenio_pidstore.providers.datacite.DataCiteMDSClient'
+    )
+
+    doi_tags = [
+        '<identifier identifierType="DOI">{doi}</identifier>',
+        ('<relatedIdentifier relatedIdentifierType="DOI" '
+         'relationType="IsVersionOf">{conceptdoi}</relatedIdentifier>'),
+    ]
+
+    # Assert calls and content
+    def assert_datacite_calls_and_content(record, doi_tags):
+        """Datacite client calls assertion helper."""
+        assert dc_mock().metadata_post.call_count == 1
+        _, doi_args, _ = dc_mock().metadata_post.mock_calls[0]
+        assert all([t.format(**record) in doi_args[0] for t in doi_tags])
+
+        assert dc_mock().doi_post.call_count == 1
+        dc_mock().doi_post.assert_any_call(
+            record['doi'],
+            'https://zenodo.org/record/{}'.format(record['doi']))
+
+    def assert_datacite_calls_with_missing_data():
+        """Datacite client calls assertion helper."""
+        assert dc_mock().metadata_post.call_count == 0
+        assert dc_mock().doi_post.call_count == 0
+
+    def create_versioned_record(recid_value, conceptrecid):
+        """Utility function for creating versioned records."""
+        recid = PersistentIdentifier.create(
+            'recid', recid_value, status=PIDStatus.RESERVED)
+        pv = PIDVersioning(parent=conceptrecid)
+        pv.insert_draft_child(recid)
+
+        record_metadata = deepcopy(minimal_record)
+        # Remove the DOI
+        del record_metadata['doi']
+        record_metadata['conceptrecid'] = conceptrecid.pid_value
+        record_metadata['recid'] = int(recid.pid_value)
+        record = ZenodoRecord.create(record_metadata)
+        zenodo_record_minter(record.id, record)
+        record.commit()
+
+        return recid, record
+
+    # Create conceptrecid for the records
+    conceptrecid = PersistentIdentifier.create(
+        'recid', '100', status=PIDStatus.RESERVED)
+
+    # Create a reserved recid
+    recid1, r1 = create_versioned_record('352543', conceptrecid)
+
+    # no registered local DOIs
+    schedule_update_datacite_metadata(1)
+    assert_datacite_calls_with_missing_data()
+
+    doi_pids = PersistentIdentifier.query.filter(
+        PersistentIdentifier.pid_value.contains('10.5072/zenodo.352543'))
```

Comment:
This would be more explicit and safe:

```python
doi_pid = PersistentIdentifier.query.filter(PersistentIdentifier.pid_value == '10.5072/zenodo.352543').one()
```